src/message_filters/__init__.py

```diff
@@ -227,27 +238,32 @@ class ApproximateTimeSynchronizer(TimeSynchronizer):
     """
     Approximately synchronizes messages by their timestamps.
 
-    :class:`ApproximateTimeSynchronizer` synchronizes incoming message filters by the
-    timestamps contained in their messages' headers. The API is the same as TimeSynchronizer
-    except for an extra `slop` parameter in the constructor that defines the delay (in seconds)
-    with which messages can be synchronized. The ``allow_headerless`` option specifies whether
-    to allow storing headerless messages with current ROS time instead of timestamp. You should
+    :class:`ApproximateTimeSynchronizer` synchronizes incoming message filters
+    by the timestamps contained in their messages' headers. The API is the same
+    as TimeSynchronizer except for an extra `slop` parameter in the constructor
+    that defines the delay (in seconds) with which messages can be synchronized.
+    The ``allow_headerless`` option specifies whether to allow storing
+    headerless messages with current ROS time instead of timestamp. You should
     avoid this as much as you can, since the delays are unpredictable.
     """
 
     def __init__(self, fs, queue_size, slop, allow_headerless=False):
         TimeSynchronizer.__init__(self, fs, queue_size)
-        self.slop = rospy.Duration.from_sec(slop)
+        self.slop = Duration(seconds=slop).nanoseconds
```

Comment:
Please use the semantic datatypes in the storage don't just convert it to the int values.

Reply:
Here `nanoseconds` is the [internal API](https://github.com/ros2/rclpy/blob/master/rclpy/rclpy/duration.py#L30-L32) of `Duration` , sorry, what does the `semantic datatypes` means from you here ? thanks !

Reply:
the `nanoseconds` call here returns the count of nanoseconds in the duration. It's a bare number. You should instead store the Duration object itself. The Duration object has the semantic meaning of representing a duration. Which can be added and subtracted to time or other duration objects, but can't be added to a distance. Whereas a bare number you can do anything you want with them and the implementer must keep track of the semantic meaning.